### PR TITLE
Add sugar for already_AddRefed

### PR DESCRIPTION
**Pull request by upsuper**
This is Servo side change of [bug 1366247](https://bugzilla.mozilla.org/show_bug.cgi?id=1366247).

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16976)

### CODE REVIEW

File: components/style/gecko_string_cache/mod.rs [hunk 1]
```diff
@@ -358,3 +340,21 @@ impl From<*mut nsIAtom> for Atom {
...
 }
+
+impl From<already_AddRefed<nsIAtom>> for Atom {
+    #[inline]
+    fn from(ptr: already_AddRefed<nsIAtom>) -> Atom {
+        let ptr = ptr.take();
+        debug_assert!(!ptr.is_null());
+        unsafe { Atom(WeakAtom::new(ptr)) }
+    }
+}
```
Comment (nox) [inline components/style/gecko_string_cache/mod.rs]: This is unsound.

```rust
Atom::from(already_AddRefed::null())
```

Either make that `debug_assert` into `assert` or mark `already_AddRefed::null` as unsafe. I would prefer the latter.

**Comment by nox** [comment]
@bors-servo r-

The `From<already_Refed<nsIAtom>>` impl is unsound.

File: components/style/gecko_string_cache/mod.rs (hunk 1 above)
Comment (upsuper) [inline components/style/gecko_string_cache/mod.rs]: Hmmm, `already_AddRefed` itself is nullable in Gecko code... I guess we should probably just make it non-nullable in Rust side.

**Pull request merged**